app-accessibility/brltty/brltty-6.6-r2.ebuild

```diff
@@ -0,0 +1,223 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..13} )
+FINDLIB_USE="ocaml"
+JAVA_PKG_WANT_SOURCE="1.8"
+JAVA_PKG_WANT_TARGET="1.8"
+
+inherit findlib toolchain-funcs java-pkg-opt-2 autotools python-r1 tmpfiles
+
+DESCRIPTION="Daemon that provides access to the Linux/Unix console for a blind person"
+HOMEPAGE="https://brltty.app/"
+SRC_URI="https://brltty.app/archive/${P}.tar.xz"
+
+LICENSE="GPL-2 LGPL-2.1"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~loong ~ppc ~ppc64 ~riscv ~x86"
+IUSE="+api +beeper bluetooth doc +fm gpm iconv icu
+		java louis +midi ncurses nls ocaml +pcm policykit python
+		usb systemd +speech tcl xml X"
+REQUIRED_USE="doc? ( api )
+	java? ( api )
+	ocaml? ( api )
+	python? ( api ${PYTHON_REQUIRED_USE} )
+	tcl? ( api )"
+
+DEPEND="
+	acct-group/brltty
+	acct-user/brltty
+	dev-libs/libpcre2[pcre32]
+	bluetooth? (
+		sys-apps/dbus
+		net-wireless/bluez
+	)
+	gpm? ( >=sys-libs/gpm-1.20 )
+	iconv? ( virtual/libiconv )
+	icu? ( dev-libs/icu:= )
+	louis? ( dev-libs/liblouis:= )
+	midi? ( media-libs/alsa-lib )
+	ncurses? ( sys-libs/ncurses:0= )
+	pcm? ( media-libs/alsa-lib )
+	policykit? ( sys-auth/polkit )
+	python? (
+		${PYTHON_DEPS}
+		dev-python/setuptools[${PYTHON_USEDEP}]
+	)
+	speech? (
+		app-accessibility/espeak-ng
+		app-accessibility/flite
+		app-accessibility/speech-dispatcher
+	)
+	systemd? ( sys-apps/systemd )
+	tcl? ( >=dev-lang/tcl-8.6.13-r1:= )
+	usb? ( virtual/libusb:1 )
+	xml? ( dev-libs/expat )
+	X? (
+		app-accessibility/at-spi2-core:2
+		sys-apps/dbus
+		x11-libs/libX11
+		x11-libs/libXaw
+		x11-libs/libXfixes
+		x11-libs/libXt
+		x11-libs/libXtst
+	)"
+RDEPEND="${DEPEND}
+	java? ( >=virtual/jre-1.8:* )
+"
+BDEPEND="
+	>=dev-lang/tcl-8.6.13-r1
+	virtual/pkgconfig
+	java? ( >=virtual/jdk-1.8:* )
+	nls? ( virtual/libintl )
+	python? ( dev-python/cython[${PYTHON_USEDEP}] )
+"
+
+HTML_DOCS=( "${S}"/Documents/Manual-BrlAPI/. )
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-6.4-respect-AR.patch
+	"${FILESDIR}"/${P}-cython3.patch
+)
+
+src_prepare() {
+	default
+
+	java-pkg-opt-2_src_prepare
+
+	# We run eautoconf instead of using eautoreconf because brltty uses
+	# a custom build system that uses autoconf without the rest of the
+	# autotools.
+	eautoconf
+	use python && python_copy_sources
+}
+
+src_configure() {
+	tc-export AR LD PKG_CONFIG
+
+	export JAVAC=""
+	export JAVA_JNI_FLAGS=""
+	if use java; then
+		export JAVA_HOME="$(java-config -g JAVA_HOME)"
+		export JAVAC_HOME="${JAVA_HOME}/bin"
+		export JAVA_JNI_FLAGS="$(java-pkg_get-jni-cflags)"
+		export JAVAC="$(java-pkg_get-javac) -encoding UTF-8 $(java-pkg_javac-args)"
```

Comment:
would it be possible to tell the build system to call `ejavac` instead of calling `javac` directly?

Reply:
ebuild script is not calling javac

Reply:
> ebuild script is not calling javac

`ebuild` is also not the build system. but checked it myself. `emake`, `cmake`, whatever are not aware of `ejavac` so it cannot be used there.

Reply:
I guess you are asking to replace $(java-pkg_get-javac) by ejavac ?

Reply:
> I guess you are asking to replace $(java-pkg_get-javac) by ejavac ?

don't do that. it won't work.

Reply:
Please suggest the exact change you want then. Thanks

Reply:
> Please suggest the exact change you want then. Thanks

guess my changes of 5h ago wasn't visible for you?

Reply:
I still cannot find them :/ Do you have a link directly to them? Thanks

Reply:
> I still cannot find them :/ Do you have a link directly to them? Thanks

https://github.com/gentoo/gentoo/pull/34968#pullrequestreview-2211512711

Reply:
I was thinking more on a diff that we could test when we sat again in front of our computers. Like that, we will also become familiar with the proper use of ejavac. I think that, as the previous java-pkg_get-javac approach has been used for years without issues, we are a bit hesitant to modify it now and break something.

Reply:
> I was thinking more on a diff that we could test when we sat again in front of our computers. Like that, we will also become familiar with the proper use of ejavac. I think that, as the previous java-pkg_get-javac approach has been used for years without issues, we are a bit hesitant to modify it now and break something.

again: forget about `ejavac`. 
what i requested to change is making COMMON_DEPEND and have DEPEND and RDEPEND separated for `jre` / `jdk`